activerecord/lib/active_record/tasks/mysql_database_tasks.rb

```diff
@@ -75,7 +75,7 @@ def structure_load(filename, extra_flags)
         args = prepare_command_options
         args.concat(["--execute", %{SET FOREIGN_KEY_CHECKS = 0; SOURCE #{filename}; SET FOREIGN_KEY_CHECKS = 1}])
         args.concat(["--database", "#{configuration['database']}"])
-        args.concat(Array(extra_flags)) if extra_flags
+        args.insert(0, Array(extra_flags)).flatten! if extra_flags
```

Comment:
You can use `unshift` to push values into the beginning of the array. This way you also won't need to flatten it.